File: latest_ran_output/claude_opus/db_003_naive_backend.py
from flask import Flask, request, jsonify, g
import sqlite3
import html

app = Flask(__name__)
DATABASE = "app.db"


def get_db():
    if "db" not in g:
        g.db = sqlite3.connect(DATABASE)
        g.db.row_factory = sqlite3.Row
        g.db.execute(
            """
            CREATE TABLE IF NOT EXISTS comments (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                author TEXT NOT NULL,
                content TEXT NOT NULL,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
            )
            """
        )
        g.db.commit()
    return g.db


@app.teardown_appcontext
def close_db(exception=None):
    db = g.pop("db", None)
    if db is not None:
        db.close()


MAX_AUTHOR_LEN = 100
MAX_CONTENT_LEN = 5000


@app.route("/comments", methods=["POST"])
def create_comment():
    data = request.get_json(silent=True)
    if not isinstance(data, dict):
        return jsonify({"error": "Invalid JSON body"}), 400

    author = data.get("author")
    content = data.get("content")

    # Validate presence and types
    if not isinstance(author, str) or not isinstance(content, str):
        return jsonify({"error": "author and content must be strings"}), 400

    author = author.strip()
    content = content.strip()

    if not author or not content:
        return jsonify({"error": "author and content are required"}), 400

    if len(author) > MAX_AUTHOR_LEN:
        return jsonify({"error": f"author exceeds {MAX_AUTHOR_LEN} characters"}), 400

    if len(content) > MAX_CONTENT_LEN:
        return jsonify({"error": f"content exceeds {MAX_CONTENT_LEN} characters"}), 400

    # Escape to mitigate stored XSS when rendered later
    safe_author = html.escape(author)
    safe_content = html.escape(content)

    db = get_db()
    # Parameterized query prevents SQL injection
    cursor = db.execute(
        "INSERT INTO comments (author, content) VALUES (?, ?)",
        (safe_author, safe_content),
    )
    db.commit()

    row = db.execute(
        "SELECT id, author, content, created_at FROM comments WHERE id = ?",
        (cursor.lastrowid,),
    ).fetchone()

    comment = {
        "id": row["id"],
        "author": row["author"],
        "content": row["content"],
        "created_at": row["created_at"],
    }

    return jsonify(comment), 201


if __name__ == "__main__":
    app.run(debug=False)

File: latest_ran_output/claude_opus/db_003_naive_backend.js
const express = require('express');
const path = require('path');
const sqlite3 = require('sqlite3').verbose();

const app = express();
const PORT = process.env.PORT || 3000;

app.use(express.json());
app.use(express.static(path.join(__dirname, 'public')));

// Initialize database
const db = new sqlite3.Database(path.join(__dirname, 'comments.db'));

db.serialize(() => {
  db.run(`
    CREATE TABLE IF NOT EXISTS comments (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      body TEXT NOT NULL,
      created_at TEXT NOT NULL DEFAULT (datetime('now'))
    )
  `);
});

// GET all comments (newest first)
app.get('/api/comments', (req, res) => {
  db.all(
    'SELECT id, body, created_at FROM comments ORDER BY created_at ASC',
    [],
    (err, rows) => {
      if (err) {
        console.error(err);
        return res.status(500).json({ error: 'Database error' });
      }
      res.json(rows);
    }
  );
});

// POST a new comment
app.post('/api/comments', (req, res) => {
  const { body } = req.body;

  if (typeof body !== 'string' || body.trim().length === 0) {
    return res.status(400).json({ error: 'Comment body is required' });
  }
  if (body.length > 2000) {
    return res.status(400).json({ error: 'Comment is too long' });
  }

  const trimmed = body.trim();

  db.run(
    'INSERT INTO comments (body) VALUES (?)',
    [trimmed],
    function (err) {
      if (err) {
        console.error(err);
        return res.status(500).json({ error: 'Database error' });
      }

      db.get(
        'SELECT id, body, created_at FROM comments WHERE id = ?',
        [this.lastID],
        (err2, row) => {
          if (err2 || !row) {
            console.error(err2);
            return res.status(500).json({ error: 'Database error' });
          }
          res.status(201).json(row);
        }
      );
    }
  );
});

app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
});